### ISSUE TITLE
Running some stages on grid and others locally

### ISSUE THREAD

**Issue by johnomics** [marbl/canu#102]
(Possibly related to #51)

We have a 64-core, 512GB RAM machine capable of submitting jobs to a cluster of 16-core, 64GB RAM nodes. If possible, I'd like to run large memory stages on the 512GB machine and the other stages on the cluster.

The [Canu tutorial](http://canu.readthedocs.org/en/stable/tutorial.html) suggests this is possible, as useGrid and gridOptions are supposed to take tags as the Memory, Threads and Concurrency options do:

```
useGrid<tag>=boolean
Run this stage on the grid, usually in parallel.
gridOptions<tag>=string
Supply this string to the grid submit command.
<tag>Memory=integer
Use this many gigabytes of memory, per process.
<tag>Threads
Use this many compute threads per process.
<tag>Concurrency
If not on the grid, run this many jobs at the same time.
```

But running with, for example, useGridmeryl=0 fails:
```ERROR:  Paramter 'useGridmeryl' is not known.```

I can't see anything in the code that processes tags like this for these options.

Is it possible to limit grid usage to certain stages, or are these options not implemented?

**Comment by prince26121991** [comment]
If I am understanding that tutorial correctly thn 
Available prefixes for tag are:
master
cns
cor
cormhap
obtmhap
utgmhap
corovl
obtovl
utgovl
cormmap
obtmmap
utgmmap
oea
ovb
ovs
red

**Comment by johnomics** [comment]
The problem is that useGrid doesn't have these tag options, although gridOptions does, so it's not clear how to avoid using the grid for certain steps - can it be done by specifying particular gridOptions?

```
$ canu -options | grep gridOptions
gridOptions                             Grid engine options applied to all jobs
gridOptionsExecutive                    Grid engine options applied to the canu executive script
gridOptionsJobName                      Grid jobs job-name suffix
gridOptionsbat                          Grid engine options applied to unitig construction jobs
gridOptionscns                          Grid engine options applied to unitig consensus jobs
gridOptionscor                          Grid engine options applied to read correction jobs
gridOptionscormhap                      Grid engine options applied to mhap overlaps for correction jobs
gridOptionscormmap                      Grid engine options applied to mmap overlaps for correction jobs
gridOptionscorovl                       Grid engine options applied to overlaps for correction jobs
gridOptionsmeryl                        Grid engine options applied to mer counting jobs
gridOptionsobtmhap                      Grid engine options applied to mhap overlaps for trimming jobs
gridOptionsobtmmap                      Grid engine options applied to mmap overlaps for trimming jobs
gridOptionsobtovl                       Grid engine options applied to overlaps for trimming jobs
gridOptionsoea                          Grid engine options applied to overlap error adjustment jobs
gridOptionsovb                          Grid engine options applied to overlap store bucketizing jobs
gridOptionsovs                          Grid engine options applied to overlap store sorting jobs
gridOptionsred                          Grid engine options applied to read error detection jobs
gridOptionsutgmhap                      Grid engine options applied to mhap overlaps for unitig construction jobs
gridOptionsutgmmap                      Grid engine options applied to mmap overlaps for unitig construction jobs
gridOptionsutgovl                       Grid engine options applied to overlaps for unitig construction jobs

$ canu -options | grep useGrid
useGrid                                 If 'true', enable grid-based execution; if 'false', run all jobs on the local machine; if 'remote', create jobs for grid execution but do not submit; default 'true'
```

**Comment by skoren** [comment]
Currently, you'd have to do this manually. Each parallel step is submitted as an array and the executive job is submitted with a hold on the array job. If you cancel the holding job and just run canu on the command line when the array job competes with useGrid=0 it will keep running on the high-memory node until you kill it and re-run with useGrid=true in which case it will submit the next step to the grid. This hasn't been extensively tested so you may encounter some issues.

That said, Canu can assemble most genomes in <64gb of ram. Unless you genome is >2g, you should be fine running only on the grid. Canu will warn you if you run it and it does not have sufficient resources for your genome/data size.

**Comment by brianwalenz** [comment]
Idle.  Open new issue (or reopen this one) if needed.
